core/src/main/java/org/ehcache/config/ResourcePoolsBuilder.java

```diff
@@ -38,6 +38,23 @@ public static ResourcePoolsBuilder newResourcePoolsBuilder() {
     return new ResourcePoolsBuilder();
   }
 
+  public static ResourcePoolsBuilder newResourcePoolsBuilder(ResourcePools pools) {
+    ResourcePoolsBuilder poolsBuilder = new ResourcePoolsBuilder();
+    if (pools.getPoolForResource(ResourceType.Core.HEAP) != null) {
```

Comment:
Why hardcode the resources you consider? You should leverage the `org.ehcache.config.ResourcePools#getResourceTypeSet` method that you added.